src/audio/base_fw.c

```diff
@@ -147,19 +148,11 @@ static int basefw_hw_config(uint32_t *data_offset, char *data)
 	tlv_value_uint32_set(tuple, IPC4_TOTAL_PHYS_MEM_PAGES_HW_CFG, value);
 
 	tuple = tlv_next(tuple);
-	tlv_value_uint32_set(tuple, IPC4_HP_EBB_COUNT_HW_CFG, PLATFORM_HPSRAM_EBB_COUNT);
 
-	tuple = tlv_next(tuple);
-	/* 2 DMIC dais */
-	value =  DAI_NUM_SSP_BASE + DAI_NUM_HDA_IN + DAI_NUM_HDA_OUT +
-			DAI_NUM_ALH_BI_DIR_LINKS + 2;
-	tlv_value_uint32_set(tuple, IPC4_GATEWAY_COUNT_HW_CFG, value);
-
-	tuple = tlv_next(tuple);
-	tlv_value_uint32_set(tuple, IPC4_LP_EBB_COUNT_HW_CFG, PLATFORM_LPSRAM_EBB_COUNT);
+	/* add platform specific tuples */
+	platform_basefw_hw_config(&plat_data_offset, (char *)tuple);
```

Comment:
doesn't this break non-Intel platforms? Do you need a `weak` version? Or is this all Intel-only?

Reply:
@lyakh This is just like all functions in platform.h, all platforms need to provide one. Let me refresh the series with "posix" platform support, I missed that and that causes some of the CI checks to fail.

The one design open I had (and still have to some extent), is whether to make this a "bigger thing", e.g. something like a "platform/ipc-data.h". But I ended up not doing that and just putting thees in top-level platform.h as the longterm goal is to minimize platform interface (drop all platform/clk.h, platfrom/dma.h as these can all be handled via Zephyr interfaces, we just can't drop them yet as some platfroms still use XTOS). Especially as @lgirdwood is still not convinced we need platform.h at all (see https://github.com/thesofproject/sof/issues/7248).

One alternative would be just to add src/ipc/ipc4/basefw_intel.c and not try to abstract this now. I'd still want to keep base_fw.c clean of vendor specific parts, so it is easier to use IPC4 for all SOF targets.